packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -155,35 +156,35 @@ export class NftDetectionController extends StaticIntervalPollingControllerV1<
 
   private getOwnerNftApi({
     address,
-    offset,
+    next,
   }: {
     address: string;
-    offset: number;
+    next?: string;
   }) {
-    return `${OPENSEA_PROXY_URL}/assets?owner=${address}&offset=${offset}&limit=50`;
+    return `${OPENSEA_PROXY_URL}/chain/ethereum/account/${address}/nfts?limit=50&next=${
+      next ?? ''
+    }`;
   }
 
   private async getOwnerNfts(address: string) {
-    let nftApiResponse: { assets: ApiNft[] };
+    let nftApiResponse: { nfts: any[]; next?: string };
     let nfts: ApiNft[] = [];
-    let offset = 0;
-    let pagingFinish = false;
-    /* istanbul ignore if */
+    let next;
+
     do {
       nftApiResponse = await fetchWithErrorHandling({
```

Comment:
Can we assign a type to this as well? This would remove the need for `?.` checks below (e.g., it seems that `nfts` is guaranteed to show up in the response data  based on the API docs, so we can capture that in the type).

Reply:
I can remove some of the `?.` based on the docs.  Although still a little nervous based on them not being 100% accurate.  Just because `fetchWithErrorHandling` etc are typed with something, the JSON deserialization doesn't actually enforce those types.  You could still end up with missing fields at this point.